src/app/calculator/furnaces/flue-gas/flue-gas.service.ts

```diff
@@ -0,0 +1,373 @@
+  }
+
+  initModification() {
+    let currentBaselineData: FlueGas = this.baselineData.getValue();
+    let currentBaselineCopy = JSON.parse(JSON.stringify(currentBaselineData));
+    let modification: FlueGas;
+    if (currentBaselineCopy.flueGasType == 'By Mass') {
+      modification = {
+        flueGasByVolume: undefined,
+        flueGasByMass: currentBaselineCopy.flueGasByMass,
+        flueGasType: currentBaselineCopy.flueGasType,
+        name: undefined
+      };
+    } else {
+      modification = {
+        flueGasByVolume: currentBaselineCopy.flueGasByVolume,
+        flueGasByMass: undefined,
+        flueGasType: currentBaselineCopy.flueGasType,
+        name: undefined
+      };
+    }
+    this.modificationData.next(modification);
+  }
+
+  generateExampleData(settings: Settings) {
+    let exampleCombAirTemp: number = 80;
+    let exampleFlueGasTemp: number = 900;
+    let exampleFuelTemp: number = 80;
+    if(settings.unitsOfMeasure != 'Imperial'){
+      exampleCombAirTemp = this.convertUnitsService.value(exampleCombAirTemp).from('F').to('C');
+      exampleCombAirTemp = Number(exampleCombAirTemp.toFixed(2));
+
+      exampleFlueGasTemp = this.convertUnitsService.value(exampleFlueGasTemp).from('F').to('C');
+      exampleFlueGasTemp = Number(exampleFlueGasTemp.toFixed(2));
+
+      exampleFuelTemp = this.convertUnitsService.value(exampleFuelTemp).from('F').to('C');
+      exampleFuelTemp = Number(exampleFuelTemp.toFixed(2));
+    }
+    let exampleBaseline: FlueGas = {
+      flueGasByMass: undefined,
+      flueGasByVolume: {
+        C2H6: 8.5,
+        C3H8: 0,
+        C4H10_CnH2n: 0,
+        CH4: 87,
+        CO: 0,
+        CO2: 0.4,
+        H2: 0.4,
+        H2O: 0,
+        N2: 3.6,
+        O2: 0.1,
+        SO2: 0,
+        combustionAirTemperature: exampleCombAirTemp,
+        excessAirPercentage: 15,
+        flueGasTemperature: exampleFlueGasTemp,
+        fuelTemperature: exampleFuelTemp,
+        gasTypeId: 1,
+        o2InFlueGas: 2.857,
+        oxygenCalculationMethod: "Excess Air",
+        heatInput: 15,
+      },
+      flueGasType: 'By Volume',
+      name: 'Baseline Flue Gas'
+    }
+
+    let exampleMod: FlueGas = {
+      flueGasByMass: undefined,
+      flueGasByVolume: {
+        C2H6: 10.5,
+        C3H8: 0,
+        C4H10_CnH2n: 0,
+        CH4: 63,
+        CO: 0,
+        CO2: 0.4,
+        H2: 0.4,
+        H2O: 0,
+        N2: 3.6,
+        O2: 0.1,
+        SO2: 0,
+        combustionAirTemperature: exampleCombAirTemp,
+        excessAirPercentage: 10,
+        flueGasTemperature: 250,
+        fuelTemperature: exampleFuelTemp,
+        gasTypeId: 1,
+        o2InFlueGas: 3.124,
+        oxygenCalculationMethod: "Excess Air",
+        heatInput: 15,
+      },
+      flueGasType: 'By Volume',
+      name: 'Modification Flue Gas'
+    }
+    this.baselineData.next(exampleBaseline);
+    this.modificationData.next(exampleMod);
+
+  }
+
+  setByMassWarnings(tmpLoss: FlueGasByMass) {
```

Comment:
Adding these functions just to call the other service function seems unneeded. Just call the flueGasLossesService directly where needed